b2/src/test/java/org/jclouds/b2/features/MultipartApiMockTest.java

```diff
@@ -0,0 +1,356 @@
+            .modules(modules)
+            .buildApi(new TypeToken<B2Api>(getClass()) {});
+   }
+
+   public B2Api api(String uri, String provider) {
+      return api(uri, provider, new Properties());
+   }
+
+   public static MockWebServer createMockWebServer() throws IOException {
+      MockWebServer server = new MockWebServer();
+      server.play();
+      URL url = server.getUrl("");
+      return server;
+   }
+
+   public void assertAuthentication(MockWebServer server) {
+      assertThat(server.getRequestCount()).isGreaterThanOrEqualTo(1);
+      try {
+         assertThat(server.takeRequest().getRequestLine()).isEqualTo("GET /b2api/v1/b2_authorize_account HTTP/1.1");
+      } catch (InterruptedException e) {
+         throw Throwables.propagate(e);
+      }
+   }
+
+   /**
+    * Ensures the request has a json header for the proper REST methods.
+    *
+    * @param request
+    * @param method
+    *           The request method (such as GET).
+    * @param path
+    *           The path requested for this REST call.
+    * @see RecordedRequest
+    */
+   public void assertRequest(RecordedRequest request, String method, String path) {
+      assertThat(request.getMethod()).isEqualTo(method);
+      assertThat(request.getPath()).isEqualTo(path);
+   }
+
+   /**
+    * Ensures the request is json and has the same contents as the resource
+    * file provided.
+    *
+    * @param request
+    * @param method
+    *           The request method (such as GET).
+    * @param resourceLocation
+    *           The location of the resource file. Contents will be compared to
+    *           the request body as JSON.
+    * @see RecordedRequest
+    */
+   public void assertRequest(RecordedRequest request, String method, String path, String resourceLocation) {
+      assertRequest(request, method, path);
+      assertContentTypeIsJson(request);
+      JsonParser parser = new JsonParser();
+      JsonElement requestJson;
+      try {
+         requestJson = parser.parse(new String(request.getBody(), Charsets.UTF_8));
+      } catch (Exception e) {
+         throw Throwables.propagate(e);
+      }
+      JsonElement resourceJson = parser.parse(stringFromResource(resourceLocation));
+      assertThat(requestJson).isEqualTo(resourceJson);
+   }
+
+   /**
+    * Ensures the request has a json header.
+    *
+    * @param request
+    * @see RecordedRequest
+    */
+   private void assertContentTypeIsJson(RecordedRequest request) {
+      assertThat(request.getHeaders()).contains("Content-Type: application/json");
+   }
+
+   /**
+    * Get a string from a resource
+    *
+    * @param resourceName
+    *           The name of the resource.
+    * @return The content of the resource
+    */
+   public String stringFromResource(String resourceName) {
+      try {
+         return Strings2.toStringAndClose(getClass().getResourceAsStream(resourceName));
+      } catch (IOException e) {
+         throw Throwables.propagate(e);
+      }
+   }
```

Comment:
Create a base class for all mock tests and move the common logic there: api creation, common assertions, etc.